tools/test/audio/comp_run.sh

```diff
@@ -123,10 +125,16 @@ parse_args "$@"
 
 # Paths
 HOST_ROOT=../../testbench/build_testbench
-HOST_EXE=$HOST_ROOT/install/bin/testbench
 HOST_LIB=$HOST_ROOT/sof_ep/install/lib
 TPLG_LIB=$HOST_ROOT/sof_parser/install/lib
 TPLG_DIR=../../build_tools/test/topology
+if [[ -z $XTRUN ]]; then
+    export LD_LIBRARY_PATH=$HOST_LIB:$TPLG_LIB
+    HOST_EXE="$HOST_ROOT/install/bin/testbench"
+else
+    HOST_EXE="$XTRUN ../../testbench/build_xt_testbench/install/bin/testbench"
```

Comment:
Define HOST_ROOT and VALGRIND conditionally based on XTRUN. Then you can define HOST_EXE unconditionally:
```
HOST_EXE="$XT_RUN $HOST_ROOT/install/bin/testbench"
```

Reply:
I've changed it now. Not exactly what you proposed but something I hope is better than previous.